src/test/java/testsuite/integration/utility/ContainerHelper.java

```diff
@@ -79,20 +80,43 @@ public void runTest(GenericContainer<?> container, String task) throws IOExcepti
     assertEquals(0, exitCode, "Some tests failed.");
   }
 
+  public void debugTest(GenericContainer<?> container, String task) throws IOException, InterruptedException {
+    System.out.println("==== Container console feed ==== >>>>");
+    Consumer<OutputFrame> consumer = new ConsoleConsumer();
+    Long exitCode = execInContainer(container, consumer, "./gradlew", task, "--debug-jvm");
+    System.out.println("==== Container console feed ==== <<<<");
+    assertEquals(0, exitCode, "Some tests failed.");
+  }
+
   public GenericContainer<?> createTestContainer(String dockerImageName) {
     return createTestContainer(dockerImageName, TEST_CONTAINER_IMAGE_NAME);
   }
 
   public GenericContainer<?> createTestContainer(String dockerImageName, String testContainerImageName) {
-    return new GenericContainer<>(
+    class FixedExposedPortContainer<SELF extends GenericContainer<SELF>> extends GenericContainer<SELF> {
+
+      public FixedExposedPortContainer(ImageFromDockerfile withDockerfileFromBuilder) {
+        super(withDockerfileFromBuilder);
+      }
+
+      public SELF withFixedExposedPort(int hostPort, int containerPort) {
+        super.addFixedExposedPort(hostPort, containerPort, InternetProtocol.TCP);
+
+        return self();
+      }
```

Comment:
If the goal is to use this method `withFixedExposedPort`
![image](https://user-images.githubusercontent.com/64801825/152738150-6df62601-a41a-4146-b3fe-d303ceb1e22d.png)
Can we just use the pre-existing class `FixedHostPortGenericContainer`?

Reply:
I agree. Existing FixedHostPortGenericContainer class do the same. I think it's essential to expose port on both level 1) on container image and 2) on container level.

Reply:
`FixedHostPortGenericContainer` is deprecated and is missing the constructor we use.

Reply:
It is deprecated because it risks conflicted ports, an issue we will still have even if we create our own class. I prefer we stick to FixedHostPortGenericContainer even if it is deprecated to avoid workarounds that we will have to maintain down the line. If they do actually remove it we can just switch to their recommended alternative or find a better solution then.

Reply:
Still missing the constructor we are using. Passing in the image name will try to grab from a remote repository and since we are creating the image, it wouldn't work.